src/main/java/fi/metatavu/soteapi/wnspusher/WnsPusherNotificationController.java

```diff
@@ -0,0 +1,153 @@
+package fi.metatavu.soteapi.wnspusher;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import fi.metatavu.soteapi.settings.SystemSettingController;
+import fi.metatavu.soteapi.wnspusher.xml.Binding;
+import fi.metatavu.soteapi.wnspusher.xml.Text;
+import fi.metatavu.soteapi.wnspusher.xml.Toast;
+import fi.metatavu.soteapi.wnspusher.xml.Visual;
+
+@ApplicationScoped
+public class WnsPusherNotificationController {
+  
+  @SuppressWarnings("cdi-ambiguous-dependency")
+  @Inject
+  private SystemSettingController systemSettingController;
+  
+  @SuppressWarnings("cdi-ambiguous-dependency")
+  @Inject
+  private Logger logger;
+  
+  private Supplier<CloseableHttpClient> httpClientSupplier = HttpClients::createDefault;
+  
+  public WnsPusherNotificationController() {
+  }
+  
+  public WnsPusherNotificationController(
+      SystemSettingController systemSettingController,
+      Logger logger,
+      Supplier<CloseableHttpClient> httpClientSupplier
+  ) {
+    this.systemSettingController = systemSettingController;
+    this.logger = logger;
+    this.httpClientSupplier = httpClientSupplier;
+  }
+
+  /**
+   * Sends push notification
+   * 
+   * @param topic target topic
+   * @param title title
+   * @param message message
+   * @param data message data
+   */
+  public void sendNotificationToTopic(String topic, String title, String message, Map<String, Object> data) {
+    sendNotification(topic, title, message, data);
+  }
+
+  /**
+   * Sends push notification
+   * 
+   * @param topic target topic
+   * @param title title
+   * @param message message
+   */
+  public void sendNotificationToTopic(String topic, String title, String message) {
+    sendNotification(topic, title, message, null);
+  }
+
+  private void sendNotification(String to, String title, String message, Map<String, Object> data) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "to". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)